operator/pkg/name/name.go

```diff
@@ -101,6 +117,18 @@ func (cn ComponentName) IsAddon() bool {
 // IsComponentEnabledInSpec reports whether the given component is enabled in the given spec.
 // IsComponentEnabledInSpec assumes that controlPlaneSpec has been validated.
 func IsComponentEnabledInSpec(componentName ComponentName, controlPlaneSpec *v1alpha1.IstioOperatorSpec) (bool, error) {
+	// for addon components, enablement is defined in values part.
```

Comment:
This seems to try to turn off addons but something must be wrong. `--set values.prometheus.enabled=false` does not do anything

Reply:
this PR can only address the enablement issue of istio components now, for addons, I added a TODO to address that after https://github.com/istio/istio/issues/20316 is resolved